influxql/ast.go

```diff
@@ -642,6 +642,36 @@ type SelectStatement struct {
 	FillValue interface{}
 }
 
+// HasDerivative returns true if one of the field in the statement is a
+// derivative aggregate
+func (s *SelectStatement) HasDerivative() bool {
```

Comment:
@otoolep The same check is used in several places so I put it on the statement since it has all the fields already.  It's needed because derivative is not implemented as a map/reduce function and is a transformation run after the map/reduce jobs.  Need to be able to check when a derivative should be run.

@dgnorton Yeah, probably a good idea.  Will fix.